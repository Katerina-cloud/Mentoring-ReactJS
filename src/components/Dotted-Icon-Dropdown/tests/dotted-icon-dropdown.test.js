import React from 'react';
import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import { DottedIconDropdown } from '../';

describe('DottedIconDropdown', () => {
  it('snapshot', () => {
    const tree = render(<DottedIconDropdown />);
    expect(tree).toMatchSnapshot();
  });

  // it('show dropdown on button click', () => {
  //   const showDropdown = jest.fn();
  //   render(<DottedIconDropdown />);

  //   const icon = screen.queryByRole('toggle-dropdown');
  //   console.log(icon);
  //   userEvent.click(icon);
  //   expect(showDropdown).toHaveBeenCalledTimes(1);
  // });
});
