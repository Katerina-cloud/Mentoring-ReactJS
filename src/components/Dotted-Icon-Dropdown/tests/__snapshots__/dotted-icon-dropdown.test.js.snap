// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DottedIconDropdown snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dotted-icon-dropdown"
      >
        <div
          class="dotted-icon-dropdown__icon"
        >
          <svg
            fill="none"
            height="44"
            viewBox="0 0 45 44"
            width="45"
          >
            <ellipse
              cx="22.5"
              cy="22"
              fill="#232323"
              rx="22.5"
              ry="22"
            />
            <circle
              cx="21.5"
              cy="29.5"
              fill="#fff"
              r="2.5"
            />
            <circle
              cx="21.5"
              cy="21.5"
              fill="#fff"
              r="2.5"
            />
            <circle
              cx="21.5"
              cy="13.5"
              fill="#fff"
              r="2.5"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dotted-icon-dropdown"
    >
      <div
        class="dotted-icon-dropdown__icon"
      >
        <svg
          fill="none"
          height="44"
          viewBox="0 0 45 44"
          width="45"
        >
          <ellipse
            cx="22.5"
            cy="22"
            fill="#232323"
            rx="22.5"
            ry="22"
          />
          <circle
            cx="21.5"
            cy="29.5"
            fill="#fff"
            r="2.5"
          />
          <circle
            cx="21.5"
            cy="21.5"
            fill="#fff"
            r="2.5"
          />
          <circle
            cx="21.5"
            cy="13.5"
            fill="#fff"
            r="2.5"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
